build/configure/app-configure-step.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var utils = require('utils');
+
+var DependencyGraph = require('./dependency-graph');
+var BuildConfig = require('./build-config');
+
+// Default building steps before app's default building steps.
+// TODO: We should have a module to map these steps with different app's own
+//       config data, so that app could easily to add/remove/reorder steps for
+//       different situations.
+const POST_APP_BUILD_STEPS = [
+  'media-resolution',
+  'post-manifest',
+  'multilocale',
+  'copy-build-stage-data',
+  'webapp-optimize',
+  'webapp-zip'
+];
+
+/**
+ * AppConfigureStep would generate all the data for build back-end, ideally it
+ * should only manage the steps relating to app itself, like app's own
+ * buildscript and POST_APP_BUILD_STEPS.
+ *
+ * @param {object} options - BUILD_CONFIG from main Makefile.
+ * @param {string} appDir - the app's source dir path.
+ * @constructor
+ */
+var AppConfigureStep = function(options, appDir) {
+  this.appDir = appDir;
+  this.webapp = utils.getWebapp(appDir, options);
+  this.name = this.webapp.sourceDirectoryName;
+  this.options = options;
+  this.buildConfig = new BuildConfig();
+};
+
+AppConfigureStep.prototype = {
+  /**
+   * Configure start from here.
+   */
+  start: function() {
+    var options = this.options;
+    this._preSteps = [];
```

Comment:
nit: either all members are fixed with underscore, or no fixes for them all, unless you're exposing these things to next stage (but I don't see so). And, even though the member would only be initialized at this function, please put it in the constructor, so people could know how many members are in this instance.